hetmech/bulk_computation.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+import pandas
+import scipy.special
+import scipy.stats
+
+import hetmech.degree_group
+import hetmech.degree_weight
+import hetmech.hetmat
+
+
+def compute_save_dwpc(graph, metapath, damping=0.5, dense_threshold=1, dtype='float64', approx_ok=True,
+                      return_result=True):
+    """
+    Returns and saves a DWPC matrix. If already computed and return_result=True, will return saved result.
+    Automatically ignores inverses, corrupt files, and previously existing files.
+    """
+    metapath = graph.metagraph.get_metapath(metapath)
+    for inverse in (True, False):
+        if inverse:
+            mp = metapath.inverse
+        try:
+            return graph.read_path_counts(mp, 'dwpc', damping)
```

Comment:
Should `read_path_counts` handle reading the inverse?

Reply:
My thinking is this: while `read_path_counts` can automatically cycle through the inverses and `dwwc` matrices on disk, the problem is that we can't know what path attempted a load which failed for a reason other than `FileNotFoundError`. These are the cases where we want to delete the corrupt file and replace it with a re-computation.

Reply:
One thing we could do is have `get_path_counts_path` return the path to the file that _would_ be loaded using `read_path_counts`. This way, we could pinpoint where the file read is failing.